platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/manager/internal/AdHocThreadManager.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.threading.manager.internal;
+
+import com.swirlds.common.threading.framework.config.MultiQueueThreadConfiguration;
+import com.swirlds.common.threading.framework.config.QueueThreadConfiguration;
+import com.swirlds.common.threading.framework.config.QueueThreadPoolConfiguration;
+import com.swirlds.common.threading.framework.config.StoppableThreadConfiguration;
+import com.swirlds.common.threading.framework.config.ThreadConfiguration;
+import com.swirlds.common.threading.interrupt.InterruptableRunnable;
+import com.swirlds.common.threading.manager.ThreadBuilder;
+import com.swirlds.common.threading.manager.ThreadManager;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A simple thread manager. The goal of this implementation is to create threads without complaining about lifecycle.
+ * Eventually, this implementation should not be used in production code.
+ */
+public final class AdHocThreadManager extends AbstractThreadManager implements ThreadBuilder, ThreadManager {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public @NonNull Thread buildThread(@NonNull final Runnable runnable) {
+        return new Thread(runnable);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public ExecutorService createCachedThreadPool(
+            @NonNull final String name, @NonNull final UncaughtExceptionHandler uncaughtExceptionHandler) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(uncaughtExceptionHandler);
+        return Executors.newCachedThreadPool(buildThreadFactory(name, uncaughtExceptionHandler));
```

Comment:
For all of these, could we please use `new ThreadPoolExecutor` (or `new ScheduledThreadPoolExecutor`) to pass in both the thread factory and a bounded queue, instead of `Executors` convenience methods in this class?  Having all ExecutorService returns from this class be bounded (instead of possibly growing without limit) greatly increases the stability of the system in high load situations, and it's highly unlikely we actually need queue sizes over a few hundred entries (perhaps offer an overload for Cached or Scheduled thread pools with a queue size parameter for the rare exception).